MovingBlocks/Terasology
# feat: prevent serialisation of private component fields

### Contains

~~This pull request adds a new `ComponentTypeHandlerFactory` type specifcally for serialising components. It is mostly the same as `ObjectFieldMapTypeHandlerFactory` (from `TypeHandlerFactory`) but with an additional check for private fields. It might be possible to merge the two classes instead, this is just a first attempt to see how it behaves.~~

~~It is not possible to exclude private field serialiation from `ObjectFieldMapTypeHandlerFactory` completely since NUI depends on this behaviour extensively for UI widgets.~~

This pull request removes the ability to serialise private fields in `TypeHandlerLibrary`, apart from in a few special cases (mainly NUI). It changes the generic `ObjectFieldMapHandler` to use getter and setter methods where available instead.

### How to test

- Create a new game
- Play around for a bit (destroying blocks, loading chunks, etc.)
- Save the game
- Restart the game and load that save
- The world and surrounding entities should be similar to how you left it
- There should also not be any additional exceptions thrown due to these changes
